release_tester/license_manager_tests/license_manager_base_test_suite.py

```diff
@@ -0,0 +1,193 @@
+        self.base_cfg = installer_base_config
+        package_type = ".tar.gz" if installer_base_config.zip_package else ".deb/.rpm/NSIS"
+        self.suite_name = f"Licence manager test suite: ArangoDB v. {str(new_version)} ({package_type})"
+        self.auto_generate_parent_test_suite_name = False
+        super().__init__(child_classes=child_classes)
+        self.use_subsuite = True
+        run_props = RunProperties(
+            enterprise=True,
+            encryption_at_rest=False,
+            ssl=False,
+        )
+        self.installer_set = create_config_installer_set(
+            versions=[new_version], base_config=self.base_cfg, deployment_mode="all", run_properties=run_props
+        )
+        self.installer = self.installer_set[0][1]
+        self.starter = None
+        self.instance = None
+        self.runner = None
+        self.passvoid = "license_manager_tests"
+        self.publicip = installer_base_config.publicip
+
+    # pylint: disable=no-self-use
+    def init_child_class(self, child_class):
+        """initialise the child class"""
+        return child_class(self.new_version, self.base_cfg)
+
+    @run_after_suite
+    def shutdown(self):
+        """shutdown instance(s)"""
+        self.runner.starter_shutdown()
+
+    def add_crash_data_to_report(self):
+        """save data dir and logs in case a test failed"""
+        self.save_log_file()
+        self.save_data_dir()
+
+    def save_log_file(self):
+        """add log file to the report"""
+        if self.installer.instance and self.installer.instance.logfile.exists():
+            log = open(self.installer.instance.logfile, "r").read()
+            attach(log, "Log file " + str(self.installer.instance.logfile))
+
+    def save_data_dir(self):
+        """add datadir archive to the report"""
+        data_dir = self.installer.cfg.dbdir
+        if data_dir.exists():
+            archive = shutil.make_archive("datadir", "bztar", data_dir, data_dir)
+            attach.file(archive, "data directory archive", "application/x-bzip2", "tar.bz2")
+
+    @step
+    def check_that_license_is_not_expired(self, time_left_threshold=0):
+        """check that license is not expired"""
+        # pylint: disable=W0622
+        license = self.get_license()
+        license_expiry_timestamp = license["features"]["expires"]
+        time_left = license_expiry_timestamp - time()
+        if time_left >= 0:
+            message = f"License expires in {time_left} seconds."
+        else:
+            message = f"License expired {-1*time_left} seconds ago."
+        assert (
+            time_left > time_left_threshold
+        ), f"{message} Expected time left: more than {time_left_threshold} seconds."
+
+    # pylint: disable=R0913
+    def send_request(self, method, url, data=None, headers={}, timeout=None, instance_type=None):
+        """send HTTP request to an instance"""
+        if not instance_type:
+            if isinstance(self.runner, LeaderFollower):
```

Comment:
I would rather like to have this as a method in the respective derivate classes instead of a rumpelstilskin lookup, and an abstract empty method here.